### PR TITLE
BookStore API In progress

### PR DESCRIPTION
**Pull request by kmb4577**
Tested createBook, updateBook, getBook, findOrCreatePublisher, & findOrCreateAuthor. Continuing with testing other functions to verify functionality now.

### CODE REVIEW

**Comment by roda0313** [comment]
There are way too many conflicts in this to merge, please fix.

File: API/API.php [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-
 // initial result of the api
 $result = "An error has occurred";
 
 // needed globals
 $errorLogFile = "errors.txt";
-$databaseFile = getcwd(). "/../Database/SWEN344DB.db";
+
+//$databaseFile = getcwd(). "/../Database/SWEN344DB.db";
+$databaseFile = __DIR__.DIRECTORY_SEPARATOR."../Database/SWEN344DB.db";
```
Comment (roda0313) [inline API/API.php]: Don't change this line. It works on the VM where the API is hosted as it currently is and there is no reason to risk breaking the API with this change

File: API/API.php [hunk 2]
```diff
@@ -23,16 +24,17 @@ function general_switch()
 
 	if (isset($_GET["function"]) && in_array($_GET["function"], $possible_function_url))
 	{
+		echo "change";
```
Comment (roda0313) [inline API/API.php]: remove this line please

File: API/API.php [hunk 3]
```diff
@@ -122,93 +124,93 @@ function encrypt($string)
...
+
+		$query1->bindParam(':username', $username);
+		$query1->bindParam(':password', encrypt($password));
+		$query1->bindParam(':fname', $fname);
 		$query1->bindParam(':lname', $lname);
 		$query1->bindParam(':email', $email);
 		$query1->bindParam(':role', $role);
-		
-		$query1->execute();	
-
```
Comment (roda0313) [inline API/API.php]: Why are you adding/removing lots of spaces and tabs?

File: API/Bookstore.php [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+        {
+            return $exception->getMessage();
+        }
+        logError($exception);
+    }
+
+}
+
+
+?>
```
Comment (roda0313) [inline API/Bookstore.php]: If you wish the put this in a separate file that is fine but you need to make sure you then include this PHP file in the main PHP file. I don't see the include statement anywhere

File: API/Bookstore.php [hunk 5]
```diff
@@ -0,0 +1,525 @@
...
+	}
+	catch (Exception $exception)
+	{
+		if ($GLOBALS ["sqliteDebug"])
+		{
+			return $exception->getMessage();
+		}
+		logError($exception);
+	}
+	return $auth_id["ID"];
```
Comment (ShaydeNofziger) [inline API/Bookstore.php]: You should move this inside the try block

File: API/Bookstore.php [hunk 6]
```diff
@@ -0,0 +1,525 @@
...
+		}
+		catch (Exception $exception)
+		{
+			if ($GLOBALS ["sqliteDebug"])
+			{
+				return $exception->getMessage();
+			}
+			logError($exception);
+	}
+	return $result;
```
Comment (ShaydeNofziger) [inline API/Bookstore.php]: $result should be returned from the scope it was created in, the try block.

File: API/Bookstore.php [hunk 7]
```diff
@@ -0,0 +1,525 @@
...
+	}
+	catch (Exception $exception)
+	{
+		if ($GLOBALS ["sqliteDebug"])
+		{
+			return $exception->getMessage();
+		}
+		logError($exception);
+	}
+	return $book_result;
```
Comment (ShaydeNofziger) [inline API/Bookstore.php]: return within try block

**Comment by roda0313** [comment]
I fixed all the issues and will now merge

**Pull request merged**